tests/unit/base/MockObject.h

```diff
@@ -316,6 +316,13 @@ namespace shogun
 			watch_method("some_method", &CMockObject::some_method);
 		}
 
+		virtual CSGObject* create_empty() const
```

Comment:
curios: why did you add this here

Reply:
Basically, since `observe()` emit a clone of the given value, we happens to call `MockObject->clone()`. However, the standard `SGObject` implementation of `clone()` assumes that there is a corresponding `create()` method, for that element, inside `class_list.h`. This is not true for `MockObject` since it is used only for testing purpose. Therefore, I needed to override `create_empty` such to return a new instance without relying on the `class_list.h` methods.

Reply:
ah yep, I remember that stuff now. thanks!